src/hardware/gameblaster.cpp

```diff
@@ -133,7 +131,7 @@ void GameBlaster::Open(const int port_choice, const std::string &card_choice,
 	is_open = true;
 }
 
-bool GameBlaster::RenderOnce()
+bool GameBlaster::MaybeRenderFrame(const FrameSink sink)
```

Comment:
This interface is admittedly clunky.

A tidier interface would have this function return the "maybe-rendered" frame.

Then it would be the job of the caller to place the frame into the desired sink, be it the audio channel or the fifo queue.

However, that adds a some boiler plate frame handling and re-packaging code around  both callers of this function (passing a frame reference in), which makes the calling code not as tidy.

The audio devices have different frame interfaces; some want pointers to pointers while others want individual floats, etc. So the frame-boiler plate would look a bit different in each audio class too (another slight downside).

Open to suggestions or things I can try - happy to push delta commits so we can decide on the nicest API.